### DISCUSSION TITLE
HTML completion provider overriding VS Code HTML basics completions

### DISCUSSION THREAD

**Discussion by mattclough1** [microsoft/vscode-discussions#995] [Extension Development QnA] [answered]
Hello! I'm having an issue writing a completion provider for HTML. In summary, my completion item provider is supposed to completion some class names for HTML elements. When I register it, it works, but similar existing functionality for completions provided by the builtin VS Code HTML Basics extension seems to disappear. For example:

Given the following basic HTML file
```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" href="/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <style>
      .test-class {
        color: blue;
      }
    </style>
  </head>
  <body>
    <div id="app" class=""></div>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>
```
_without_ my extension, if I type `test-` into the `class` attribute on the `div`, VS Code suggests `test-class`, because it's defined in the `style` tag in the same document. But as soon as I enable my extension, typing text in the `class` attribute with any character that matches my completion items shows the suggestions as expected, but typing `test-` no longer includes `test-class` as a suggestion. I don't see the same thing happen with a CSS completion provider. What could cause this?

**Comment by r3m0t** [comment, marked as answer]
`test-class` is a word based suggestion. This is the "fallback" way of providing completion. As soon as another completion provider gives completions, word based suggestions are not shown. It's controlled by `editor.wordBasedSuggestions` setting.

**Reply by mattclough1** [reply]
Wow, thank you. This was driving me insane.
